pkg/sql/logictest/testdata/logic_test/regrole_type_mixed

```diff
@@ -0,0 +1,16 @@
+# LogicTest: local-mixed-21.1-21.2
+
+statement error pq: type REGROLE is not supported until version upgrade is finalized
+CREATE TABLE t(x REGROLE)
+
+statement ok
+CREATE TABLE t(x STRING)
+
+statement ok
+SET enable_experimental_alter_column_type_general = true
+
+statement error pq: type REGROLE is not supported until version upgrade is finalized
+ALTER TABLE t ALTER COLUMN x TYPE REGROLE
+
+statement error pq: type REGROLE is not supported until version upgrade is finalized
```

Comment:
maybe also add tests for arrays and tuples?

Reply:
Added for arrays.